h5io/tests/test_io.py

```diff
@@ -499,3 +500,105 @@ def test_state_python_version_error(tmp_path):
         overwrite="update",
         use_state=True,
     )
+
+
+class Singleton(ABCMeta):
+    """
+    Copied from pyiron_base.
+
+    https://github.com/pyiron/pyiron_base/blob/33910343e5e6d4c8bbb5f2522ad6714ec5184ff5/pyiron_base/interfaces/singleton.py#L23
+    """
+
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        """If an instance already exists, return that one instead."""
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
```

Comment:
To keep the test minimal: Is this part necessary?

Reply:
@jan-janssen, FYI I re-introduced the singleton here. `pickle` demands that when `__reduce__() -> str`, the imported object pass an `is` comparison with the object being reduced. Because everything is in one file here and executed top-to-bottom, I think we pass this even without the singleton metaclass; however, as soon as the class and variable definition and the load are in different files this `is` test would fail, so I feel it's better pedagogy to show here the "intended" use case of a singleton.